### PR TITLE
feat(pci.project): display correctly the new location

### PR DESCRIPTION
**Pull request by selm3n**
ref: TAPC-1359

| Question         | Answer
| ---------------- | ---
| Branch?          | `develop` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #...TAPC-1359 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

### CODE REVIEW

**Review by Tsiorifamonjena — APPROVED** [review]

**Review by ppprevost — DISMISSED** [review]

**Review by SimonChaumet — COMMENTED** [review]
Shouldn't the translations go through the translation process ?

**Comment by y4nnL** [comment]
> Shouldn't the translations go through the translation process ?

Sure, they will

**Review by y4nnL — APPROVED** [review]

File: packages/manager/apps/pci-private-network/public/translations/region/Messages_fr_FR.json [hunk 1]
```diff
@@ -56,7 +58,8 @@
   "manager_components_region_location_RBX": "Europe de l'Ouest (France)",
   "manager_components_region_location_DE": "Europe centrale (Allemagne)",
   "manager_components_region_location_UK": "Europe de l'Ouest (Grande-Bretagne)",
-  "manager_components_region_location_SGP": "Asie du Sud-Est (Singapour)",
+  "manager_components_region_location_SGP": "Asie Pacifique (Singapour)",
+  "manager_components_region_location_MUM": "Asie Pacifique (Singapour)",
```
Comment (y4nnL) [inline packages/manager/apps/pci-private-network/public/translations/region/Messages_fr_FR.json:62]:
```suggestion
  "manager_components_region_location_MUM": "Asie Pacifique (Mumbai)",
```

**Review by y4nnL — CHANGES_REQUESTED** [review]

File: packages/manager/apps/pci-public-ip/public/translations/regions/Messages_fr_FR.json [hunk 2]
```diff
@@ -46,7 +48,8 @@
   "manager_components_region_location_RBX": "Europe de l'Ouest (France)",
   "manager_components_region_location_DE": "Europe centrale (Allemagne)",
   "manager_components_region_location_UK": "Europe de l'Ouest (Grande-Bretagne)",
-  "manager_components_region_location_SGP": "Asie du Sud-Est (Singapour)",
+  "manager_components_region_location_SGP": "Asie Pacifique  (Singapour)",
```
Comment (y4nnL) [inline packages/manager/apps/pci-public-ip/public/translations/regions/Messages_fr_FR.json:51]:
```suggestion
  "manager_components_region_location_SGP": "Asie Pacifique (Singapour)",
```

File: packages/manager/apps/pci-gateway/public/translations/regions/Messages_fr_FR.json [hunk 3]
```diff
@@ -46,7 +48,8 @@
   "manager_components_region_location_RBX": "Europe de l'Ouest (France)",
   "manager_components_region_location_DE": "Europe centrale (Allemagne)",
   "manager_components_region_location_UK": "Europe de l'Ouest (Grande-Bretagne)",
-  "manager_components_region_location_SGP": "Asie du Sud-Est (Singapour)",
+  "manager_components_region_location_SGP": "Asie Pacifique  (Singapour)",
```
Comment (y4nnL) [inline packages/manager/apps/pci-gateway/public/translations/regions/Messages_fr_FR.json:51]:
```suggestion
  "manager_components_region_location_SGP": "Asie Pacifique (Singapour)",
```

File: packages/manager/apps/pci-databases-analytics/src/public/translations/regions/Messages_fr_FR.json [hunk 4]
```diff
@@ -46,7 +48,8 @@
   "region_location_RBX": "Europe de l'Ouest (France)",
   "region_location_DE": "Europe centrale (Allemagne)",
   "region_location_UK": "Europe de l'Ouest (Grande-Bretagne)",
-  "region_location_SGP": "Asie du Sud-Est (Singapour)",
+  "region_location_SGP": "Asie Pacifique  (Singapour)",
```
Comment (y4nnL) [inline packages/manager/apps/pci-databases-analytics/src/public/translations/regions/Messages_fr_FR.json:51]:
```suggestion
  "region_location_SGP": "Asie Pacifique (Singapour)",
```

**Review by y4nnL — CHANGES_REQUESTED** [review]

**Review by Tsiorifamonjena — DISMISSED** [review]

**Review by Eric-ciccotti — DISMISSED** [review]

**Review by Eric-ciccotti — DISMISSED** [review]

**Review by ppprevost — APPROVED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by Eric-ciccotti — APPROVED** [review]

**Comment by SimonChaumet** [comment]
It seems that there is a lot of non-trimmed string in the translations, do we want them ?

**Review by SimonChaumet — APPROVED** [review]

**Review by Eric-ciccotti — APPROVED** [review]

**Review by SimonChaumet — DISMISSED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by Eric-ciccotti — APPROVED** [review]

File: packages/manager/apps/pci-gateway/src/api/hooks/data.ts [hunk 5]
```diff
@@ -7,29 +7,16 @@ import { TAvailableGatewayPlansResponse } from '@/api/data/gateway-plans';
...
+  const regionSubStrings = region.split('-');
+
+  const macroRegionMap = {
+    1: regionSubStrings[0].split(/(\d)/)[0],
+    2: regionSubStrings[0],
+    3: regionSubStrings[2],
+    4: regionSubStrings[2] === 'LZ' ? regionSubStrings[3] : regionSubStrings[2],
+    5: regionSubStrings[3],
+  };
+  return macroRegionMap[regionSubStrings.length] || 'Unknown_Macro_Region';
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-gateway/src/api/hooks/data.ts:19]: Maybe you should copy the code from manager-react-components, this code was removed so I suppose there was a bug with it
https://github.com/ovh/manager/blob/master/packages/manager-react-components/src/hooks/region/useTranslatedMicroRegions.tsx#L14-L25

**Review by anooparveti — APPROVED** [review]

**Review by SimonChaumet — APPROVED** [review]

**Review by fredericvilcot — APPROVED** [review]

File: packages/manager/apps/pci-gateway/src/api/hooks/data.ts [hunk 6]
```diff
@@ -7,29 +7,17 @@ import { TAvailableGatewayPlansResponse } from '@/api/data/gateway-plans';
...
+
+  const macroRegionMap = [
+    null,
+    regionSubStrings[0].split(/(\d)/)[0],
+    regionSubStrings[0],
+    regionSubStrings[2],
+    regionSubStrings[2] === 'LZ' ? regionSubStrings[3] : regionSubStrings[2],
+    regionSubStrings[3],
+  ];
+  return macroRegionMap[regionSubStrings.length] || 'Unknown_Macro_Region';
```
Comment (anooparveti) [inline packages/manager/apps/pci-gateway/src/api/hooks/data.ts:20]: **question:** Have you tested with the changes with Localzones?

Comment (selm3n) [reply packages/manager/apps/pci-gateway/src/api/hooks/data.ts:20]: Yes the same function is used in instances and works with localZones

File: packages/manager/apps/pci-private-network/src/pages/new/steps/LocalizationStep.tsx [hunk 7]
```diff
@@ -47,26 +47,17 @@ export default function LocalizationStep(): JSX.Element {
...
+
+    const macroRegionMap = [
+      null,
+      regionSubStrings[0].split(/(\d)/)[0],
+      regionSubStrings[0],
+      regionSubStrings[2],
+      regionSubStrings[2] === 'LZ' ? regionSubStrings[3] : regionSubStrings[2],
+      regionSubStrings[3],
+    ];
+    return macroRegionMap[regionSubStrings.length] || 'Unknown_Macro_Region';
```
Comment (anooparveti) [inline packages/manager/apps/pci-private-network/src/pages/new/steps/LocalizationStep.tsx:60]: Same function is repeated in multiple apps. Can you please move it to common folder and reuse?

**Review by anooparveti — CHANGES_REQUESTED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**